### PR TITLE
Issue2

### PR DESCRIPTION
**Pull request by YaourtNature83**

### CODE REVIEW

File: src/main/ContinuousIntegrationServer.java [hunk 1]
```diff
@@ -25,13 +34,24 @@ public void handle(String target,
...
         System.out.println(target);
+        
+
 
         // here you do all the continuous integration tasks
         // for example
         // 1st clone your repository
+        File localDirectory = new File("TestGitPull\\");
+    	Git git = GitConnector.cloneRepo("https://github.com/DD2480-Group26/DD2480-CI.git",localDirectory);
+    	GitConnector.gitPull(localDirectory, "test");
```
Comment (ElisabethChen) [inline src/main/ContinuousIntegrationServer.java]: should it be called 'test'?

**Review by ElisabethChen — APPROVED** [review]
Just have a question about naming but most seems great!

File: src/main/ContinuousIntegrationServer.java [hunk 2]
```diff
@@ -1,19 +1,28 @@
...
 import java.io.IOException;
  
 import org.eclipse.jetty.server.Server;
 import org.eclipse.jetty.server.Request;
 import org.eclipse.jetty.server.handler.AbstractHandler;
+import org.eclipse.jgit.api.CloneCommand;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.PullCommand;
+import org.eclipse.jgit.api.PullResult;
+import org.eclipse.jgit.merge.MergeStrategy;
```
Comment (cadye) [inline src/main/ContinuousIntegrationServer.java]: just do"import org.eclipse.jgit.*"?

File: src/main/ContinuousIntegrationServer.java (hunk 1 above)
Comment (cadye) [reply src/main/ContinuousIntegrationServer.java]: should probably be called the same as the branch name, imo. this information should be in one of the argument strings, probably 'baseRequest' or 'request'.

File: src/main/GitConnector.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	
+	public static boolean gitPull(File localRepoPath, String branch) {
+		 boolean pullSuccessful;
+		 
+		 try (Git git = Git.open(localRepoPath)) {
+		 
+			 PullCommand pull = git.pull();
+			 pull.setRemote("origin");
+			 pull.setRemoteBranchName(branch); 
+//			 pull.setStrategy(MergeStrategy.RECURSIVE);
```
Comment (cadye) [inline src/main/GitConnector.java]: commented code

File: src/main/ContinuousIntegrationServer.java [hunk 4]
```diff
@@ -25,13 +34,24 @@ public void handle(String target,
...
+        
+
 
         // here you do all the continuous integration tasks
         // for example
         // 1st clone your repository
+        File localDirectory = new File("TestGitPull\\");
+    	Git git = GitConnector.cloneRepo("https://github.com/DD2480-Group26/DD2480-CI.git",localDirectory);
+    	GitConnector.gitPull(localDirectory, "test");
+    	GitConnector.checkoutToBranch(localDirectory, "origin/test");
```
Comment (cadye) [inline src/main/ContinuousIntegrationServer.java]: same as above comment. "origin/test" shouldn't be hard-coded.

File: src/main/ContinuousIntegrationServer.java [hunk 5]
```diff
@@ -25,13 +34,24 @@ public void handle(String target,
         baseRequest.setHandled(true);
 
         System.out.println(target);
+        
+
 
         // here you do all the continuous integration tasks
         // for example
         // 1st clone your repository
+        File localDirectory = new File("TestGitPull\\");
```
Comment (cadye) [inline src/main/ContinuousIntegrationServer.java]: should remove "Test" at the start of the directory name, or rename to "temp".

**Review by cadye — CHANGES_REQUESTED** [review]
overall good, take a look at the other comments. make sure you squash and merge, and write a good commit message.

**Review by ElisabethChen — CHANGES_REQUESTED** [review]
remembered that we should write JavaDoc for all functions (e.g. functions in GitConnector)

File: src/main/ContinuousIntegrationServer.java (hunk 2 above)
Comment (YaourtNature83) [reply src/main/ContinuousIntegrationServer.java]: Why ?

File: src/main/GitConnector.java (hunk 3 above)
Comment (YaourtNature83) [reply src/main/GitConnector.java]: I'll remove this line

File: src/main/ContinuousIntegrationServer.java (hunk 2 above)
Comment (cadye) [reply src/main/ContinuousIntegrationServer.java]: less code, but doesn't matter.

**Comment by YaourtNature83** [comment]
Added JavaDoc and minor changes. I will wait for maven to initialize the branch name.

**Review by juliaBystrom — APPROVED** [review]
I am a bit late on checking out the code but it looks good to me.

File: src/main/ContinuousIntegrationServer.java (hunk 2 above)
Comment (juliaBystrom) [reply src/main/ContinuousIntegrationServer.java]: I think writting it like this makes it more clear on what is actually used. We do not do any useless imports to.

**Review by ElisabethChen — APPROVED** [review]
Good work! The code looks great. Maybe it would be good to write which parm you are describing in JavaDoc, e.g. change "@param Https link to Git Repo" --> "@param remotePath Https link to Git Repo"

**Review by cadye — APPROVED** [review]
> I will wait for maven to initialize the branch name.

not sure why you'd have to wait for maven, but I believe Julia is working on the issue you are mentioning. otherwise Gradle is working now.

**Pull request merged**